actions/approve_csr_handler.go

```diff
@@ -48,6 +52,19 @@ func (h *approveCSRHandler) Handle(ctx context.Context, action *castai.ClusterAc
 		actionIDLogField: action.ID,
 	})
 
+	if req.AllowAutoApprove != nil && *req.AllowAutoApprove {
+		go h.RunAutoApprove(ctx)
```

Comment:
Could you describe concurrency design in the comments? There are:
- goroutine
- mutex
- multiple g.Go(...)

I wonder if all this is necessary.

Reply:
we have only one go routine with watching and approving messages
but multiple handlers can start it or stop it

Reply:
and we have approved csr for the initial node join which should be approved not by Watcher